run.sh

```diff
@@ -1,23 +1,58 @@
 #!/bin/bash
-# run the specified java class (of the org.acoli.conll.rdf package) with the provided arguments
-# after calling compile.sh to ensure it's up to date.
-
-# store the path of the conll-rdf directory in HOME
-HOME=$( dirname -- "$(realpath -- "$0")");
-TARGET="$HOME/target/classes";
-CLASSPATH="$TARGET:$HOME/lib/*";
-
-if (source $HOME/compile.sh &> /dev/null;); then
-	if [ $OSTYPE = "cygwin" ]; then
-		TARGET=$(cygpath -wa -- "$TARGET");
-		CLASSPATH=$(cygpath -pwa -- "$CLASSPATH");
-	fi;
-	#could also add -Dlog4j.configuration=file:'src/main/resources/log4j.properties' for another log4j config
-	java -Dfile.encoding=UTF8 -classpath $CLASSPATH org.acoli.conll.rdf.$* | \
+#
+# compile & run, using maven, the specified java class (of the org.acoli.conll.rdf package) with the provided arguments.
+# This is intended to be the most failproof way to use conll-rdf
+# If you need better performance (e.g. faster startup), you should probably use java on a jar-with-dependencies instead.
+
+# To the unfortunate HiWi or post-graduate tasked with modifying this file after I'm gone: Good luck!
+# Please read up on Quotes in bash before modifying this file. See https://mywiki.wooledge.org/Quotes for example.
+# Also http://mywiki.wooledge.org/WordSplitting while you're at it.
+
+# Configure Bash Strict Mode
+# * set -e to exit on error
+# * set -u to fail on references to unbound variables
+# * set -o pipefail to carry the non-zero exit code of a pipeline
+# * IFS=$'\n\t' (InternalFieldSeperator) to not split arrays on spaces. Notice the $'...' syntax
+set -euo pipefail
+IFS=$'\n\t'
```

Comment:
Explanation in the comments. It would make sense to put this in the example scripts as well, and I might do that.